src/ProjectOrigin.WalletSystem.Server/Repositories/RequestStatusRepository.cs

```diff
@@ -0,0 +1,64 @@
+using ProjectOrigin.WalletSystem.Server.Models;
+using System.Threading.Tasks;
+using System;
+using System.Data;
+using Dapper;
+
+namespace ProjectOrigin.WalletSystem.Server.Repositories;
+
+public interface IRequestStatusRepository
+{
+    Task InsertRequestStatus(RequestStatus status);
+    Task<RequestStatus?> GetRequestStatus(Guid requestId);
+    Task SetRequestStatus(Guid requestId, RequestStatusState requestStatus, string? failedReason = null);
+}
+
+public class RequestStatusRepository : IRequestStatusRepository
+{
+    private readonly IDbConnection _connection;
+
+    public RequestStatusRepository(IDbConnection connection) => this._connection = connection;
+
+    public async Task InsertRequestStatus(RequestStatus status)
+    {
+        await _connection.ExecuteAsync(
+            @"INSERT INTO request_statuses(request_id, status, failed_reason)
+              VALUES (@requestId, @status, @failedReason)",
+            new
+            {
+                status.RequestId,
+                status.Status,
+                status.FailedReason
+            });
+    }
+
+    public Task<RequestStatus?> GetRequestStatus(Guid requestId)
+    {
+        return _connection.QueryFirstOrDefaultAsync<RequestStatus>(
+            @"SELECT s.*
+              FROM request_statuses s
+              WHERE s.request_id = @requestId",
```

Comment:
Should there be a owner here?
Or should it be allowed to get status on request you not the owner of?

Reply:
added owner throughout the solution